### PR TITLE
Fix : OpenMPI with Intel compiler (and similar issues)

### PR DESCRIPTION
**Pull request by alalazo**
Modifications : 
- [x] if `-v` or similar flags are present in **any** position, mode is `vcheck` 
- [x] the wrapper will not add search paths, rpaths etc. in `vcheck` mode

Refers to [this thread](https://groups.google.com/forum/#!topic/spack/hvo1h9yyhrw) in the google group

Note : please check carefully this PR, as it changes `cc` and was not tested with other compilers than Intel.

### CODE REVIEW

**Comment by alalazo** [comment]
@adamjstewart I think you may be interested in this. Please let me know if it works as expected.

**Comment by adamjstewart** [comment]
Yes, I am very interested in this. I will give it a thorough testing today and get back to you tomorrow.

File: lib/spack/env/cc [hunk 1]
```diff
@@ -330,3 +343,5 @@ if [ "$SPACK_DEBUG" = "TRUE" ]; then
 fi
 
 exec "${full_command[@]}"
+
```
Comment (eschnett) [inline lib/spack/env/cc]: These newlines seem accidental.

File: lib/spack/env/cc [hunk 2]
```diff
@@ -145,6 +153,11 @@ fi
 # Save original command for debug logging
 input_command="$@"
 
+if [ "$mode" == vcheck ] ; then
+    exec "${input_command}"
+    exit
```
Comment (eschnett) [inline lib/spack/env/cc]: You can leave out the `exit`, or if you want to keep it, you should exit with an error instead.

**Comment by tgamblin** [comment]
@adamjstewart: Let me know when you've had a chance to test this and I will get it merged in.

**Comment by adamjstewart** [comment]
I've had mixed results so far. The proposed changes do indeed solve the problem everyone was having with installing OpenMPI with Intel compilers. My Intel compilers still cannot install anything without absurdly long `icc.cfg`, `icpc.cfg`, and `ifort.cfg` files:

```
-Xlinker -rpath -Xlinker /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/compiler/lib/intel64 -Xlinker -rpath -Xlinker /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/mpi/intel64/lib -Xlinker -rpath -Xlinker /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/mpi/mic/lib -Xlinker -rpath -Xlinker /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/ipp/lib/intel64 -Xlinker -rpath -Xlinker /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/compiler/lib/intel64 -Xlinker -rpath -Xlinker /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/mkl/lib/intel64 -Xlinker -rpath -Xlinker /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/tbb/lib/intel64/gcc4.4 -Xlinker -rpath -Xlinker /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/debugger_2016/libipt/intel64/lib -Xlinker -rpath -Xlinker /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/daal/lib/intel64_lin -Xlinker -rpath -Xlinker /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/compiler/lib/intel64_lin -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/ipp/lib/intel64 -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/compiler/lib/intel64 -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/mkl/lib/intel64 -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/tbb/lib/intel64/gcc4.4 -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/daal/lib/intel64_lin -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/compiler/lib/intel64_lin
```

I've tried only passing in `LD_LIBRARY_PATH` or `LIBRARY_PATH` (`Xlinker` or `-L`) but neither work on their own. I need to use both. With these config files in place and the proposed changes in affect, I have been able to install the following packages and their dependencies:

```
spack install openmpi %intel +tm +psm
spack install openmpi %intel +tm +verbs
spack install mpich %intel +verbs
spack install mvapich2 %intel +psm
spack install szip %intel
spack install hdf %intel +szip
```

However, I was unable to install:

```
spack install mvapich2 %intel +mrail
```

at first. By sourcing `intel-16.0.1/bin/compiler_vars.sh`, I was able to install this package. There must be some other environment variable that is necessary here. The place I am currently hung up on is with HDF5. This installation:

```
spack install hdf5 %gcc +szip
```

fails for me no matter what I do. Keep in mind that all of these installations went smoothly for me with the Spack-built GCC. I haven't gotten a chance yet to see if I can install these packages outside of Spack, so it may just be an Intel problem, not a Spack problem. If I confirm that it is indeed a Spack problem, I'll post some `spack-build.out` and `config.log` files.

As for the proposed changes, they seem to address the earlier problems with Intel compilers and OpenMPI. Someone should probably make sure that these changes play well with other compilers before we merge this PR.

**Comment by adamjstewart** [comment]
It looks like the reason I am having trouble building HDF5 with Intel is that the Intel-built ZLib does not include shared object libraries (libz.so). The GCC-built ZLib comes with both archive (libz.a) and shared object (libz.so) libraries. I'm going to investigate why Spack fails to build shared libraries for ZLib with Intel. Let me know if you would like to see config.log/spack-build.out files.

**Comment by adamjstewart** [comment]
Uh oh. @alalazo it looks like your changes broke the Zlib installation. When I install Zlib with the current develop branch, everything looks good:

```bash
$ ls /soft/spack/opt/spack/linux-x86_64/intel-16.0.1/zlib-1.2.8-e7ta6ybpm5e2thhmyem7vfbrokzhki3s/lib/
libz.a  libz.so  libz.so.1  libz.so.1.2.8  pkgconfig
```

But when I install Zlib from @alalazo's fixes/intel_openmpi branch, things don't look so good:

```bash
$ ls /soft/spack/opt/spack/linux-x86_64/intel-16.0.1/zlib-1.2.8-e7ta6ybpm5e2thhmyem7vfbrokzhki3s/lib/
libz.a  pkgconfig
```

Can you reproduce this problem?

**Comment by alalazo** [comment]
@adamjstewart Thanks for taking the time to test this PR. I will be quite busy tomorrow, but I'll definitely try to investigate the issue better on Wednesday. Let me know if in the meanwhile you have further info.

**Comment by adamjstewart** [comment]
No problem. Here are some log files for you:

[configure.txt](https://github.com/LLNL/spack/files/151856/configure.txt)
[develop-configure.txt](https://github.com/LLNL/spack/files/151859/develop-configure.txt)
[develop-spack-build.txt](https://github.com/LLNL/spack/files/151858/develop-spack-build.txt)
[fixes-intel-openmpi-configure.txt](https://github.com/LLNL/spack/files/151857/fixes-intel-openmpi-configure.txt)
[fixes-intel-openmpi-spack-build.txt](https://github.com/LLNL/spack/files/151860/fixes-intel-openmpi-spack-build.txt)

**Comment by alalazo** [comment]
@adamjstewart I am not able to reproduce the problem on a RHEL 6:
```
$ spack install zlib%intel
==> Installing zlib
==> Trying to fetch from file:///ssoft/E5v3/downloads/zlib/zlib-1.2.8.tar.gz
curl: (37) Couldn't open file /ssoft/E5v3/downloads/zlib/zlib-1.2.8.tar.gz
==> Fetching from file:///ssoft/E5v3/downloads/zlib/zlib-1.2.8.tar.gz failed.
==> Trying to fetch from file:///home/culpo/mirror/download/zlib/zlib-1.2.8.tar.gz
######################################################################## 100,0%
==> Staging archive: /home/culpo/github/spack/var/spack/stage/zlib-1.2.8-qpfasakeppnp35su5gv6m2zl4kbaiqb2/zlib-1.2.8.tar.gz
==> Created stage in /home/culpo/github/spack/var/spack/stage/zlib-1.2.8-qpfasakeppnp35su5gv6m2zl4kbaiqb2.
==> No patches needed for zlib.
==> Building zlib.
==> Successfully installed zlib.
  Fetch: 0.02s.  Build: 13.01s.  Total: 13.03s.
[+] /home/culpo/github/spack/opt/spack/x86_E5v2_IntelIB/intel-15.0.2/zlib-1.2.8-qpfasakeppnp35su5gv6m2zl4kbaiqb2

$ ls /home/culpo/github/spack/opt/spack/x86_E5v2_IntelIB/intel-15.0.2/zlib-1.2.8-qpfasakeppnp35su5gv6m2zl4kbaiqb2
include  lib  share

$ ls /home/culpo/github/spack/opt/spack/x86_E5v2_IntelIB/intel-15.0.2/zlib-1.2.8-qpfasakeppnp35su5gv6m2zl4kbaiqb2/lib/
libz.a  libz.so  libz.so.1  libz.so.1.2.8  pkgconfig

$ ldd /home/culpo/github/spack/opt/spack/x86_E5v2_IntelIB/intel-15.0.2/zlib-1.2.8-qpfasakeppnp35su5gv6m2zl4kbaiqb2/lib/libz.so
	linux-vdso.so.1 =>  (0x00007fff54bff000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f2466094000)
	libimf.so => /ssoft/intel/15.0.0/RH6/all/x86_E5v2/composer_xe_2015.2.164/compiler/lib/intel64/libimf.so (0x00007f2465bd9000)
	libsvml.so => /ssoft/intel/15.0.0/RH6/all/x86_E5v2/composer_xe_2015.2.164/compiler/lib/intel64/libsvml.so (0x00007f2464d06000)
	libirng.so => /ssoft/intel/15.0.0/RH6/all/x86_E5v2/composer_xe_2015.2.164/compiler/lib/intel64/libirng.so (0x00007f2464afe000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f246487a000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f2464664000)
	libintlc.so.5 => /ssoft/intel/15.0.0/RH6/all/x86_E5v2/composer_xe_2015.2.164/compiler/lib/intel64/libintlc.so.5 (0x00007f2464408000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f2464204000)
	/lib64/ld-linux-x86-64.so.2 (0x0000003f07800000)
```

**Comment by adamjstewart** [comment]
Interesting. The reason that zlib configure thinks I'm using gcc is because `icc -v` looks like this for me:
```bash
$ icc -v
icc version 16.0.1 (gcc version 4.4.7 compatibility)
ld    /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../lib64/crt1.o /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../lib64/crti.o /usr/lib/gcc/x86_64-redhat-linux/4.4.7/crtbegin.o --eh-frame-hdr --build-id -dynamic-linker /lib64/ld-linux-x86-64.so.2 -m elf_x86_64 -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/ipp/lib/intel64 -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/compiler/lib/intel64 -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/mkl/lib/intel64 -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/tbb/lib/intel64/gcc4.4 -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/daal/lib/intel64_lin -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/compiler/lib/intel64_lin -o a.out -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/ipp/lib/intel64 -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/compiler/lib/intel64 -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/mkl/lib/intel64 -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/tbb/lib/intel64/gcc4.4 -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/daal/lib/intel64_lin -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/daal/../compiler/lib/intel64_lin -L/soft/intel/13.1.3/lib/intel64 -L/soft/intel/13.1.3/ipp/lib/intel64 -L/soft/intel/13.1.3/tbb/lib/intel64/gcc4.4 -L/soft/lcrc/lib -L/blues/gpfs/home/software/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/compiler/lib/intel64_lin -L/usr/lib/gcc/x86_64-redhat-linux/4.4.7/ -L/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../lib64 -L/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../lib64/ -L/lib/../lib64 -L/lib/../lib64/ -L/usr/lib/../lib64 -L/usr/lib/../lib64/ -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/ipp/lib/intel64/ -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/compiler/lib/intel64/ -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/mkl/lib/intel64/ -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/tbb/lib/intel64/gcc4.4/ -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/daal/lib/intel64_lin/ -L/soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/daal/../compiler/lib/intel64_lin/ -L/soft/intel/13.1.3/lib/intel64/ -L/soft/intel/13.1.3/ipp/lib/intel64/ -L/soft/intel/13.1.3/tbb/lib/intel64/gcc4.4/ -L/soft/lcrc/lib/ -L/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../ -L/lib64 -L/lib/ -L/usr/lib64 -L/usr/lib -rpath /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/compiler/lib/intel64 -rpath /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/mpi/intel64/lib -rpath /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/mpi/mic/lib -rpath /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/ipp/lib/intel64 -rpath /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/compiler/lib/intel64 -rpath /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/mkl/lib/intel64 -rpath /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/tbb/lib/intel64/gcc4.4 -rpath /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/debugger_2016/libipt/intel64/lib -rpath /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/daal/lib/intel64_lin -rpath /soft/spack/opt/spack/linux-x86_64/binaries/intel-16.0.1/compilers_and_libraries_2016.1.150/linux/compiler/lib/intel64_lin -Bdynamic -Bstatic -limf -lsvml -lirng -Bdynamic -lm -Bstatic -lipgo -ldecimal --as-needed -Bdynamic -lcilkrts -lstdc++ --no-as-needed -lgcc -lgcc_s -Bstatic -lirc -lsvml -Bdynamic -lc -lgcc -lgcc_s -Bstatic -lirc_s -Bdynamic -ldl -lc /usr/lib/gcc/x86_64-redhat-linux/4.4.7/crtend.o /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../lib64/crtn.o
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../lib64/crt1.o: In function `_start':
(.text+0x20): undefined reference to `main'
```

Configure finds the word gcc and decides I'm using gcc. This doesn't happen when using fixes/intel_openmpi and so -fPIC never gets added.

File: lib/spack/env/cc [hunk 3]
```diff
@@ -113,14 +113,22 @@ case "$command" in
...
-# Finish setting up the mode.
+# If any of the arguments below is present then the mode is vcheck. In vcheck mode nothing is added in terms of extra search paths or libraries
 if [ -z "$mode" ]; then
-    mode=ccld
     for arg in "$@"; do
         if [ "$arg" = -v -o "$arg" = -V -o "$arg" = --version -o "$arg" = -dumpversion ]; then
             mode=vcheck
             break
-        elif [ "$arg" = -E ]; then
+	fi
```
Comment (adamjstewart) [inline lib/spack/env/cc]: Can you replace this tab with spaces?

**Comment by tgamblin** [comment]
I'm confused about how this change can cause the differences @adamjstewart is seeing.  It seems like the main effect of the changes is that in @alalazo's version, checks for version arguments are done before everything else and take precedence.  Are there weird commands in the zlib build that contain both version check args *and* `-E` or `-c`?  Those would be broken by this change.

**Comment by alalazo** [comment]
@adamjstewart Comparing your configure logs with mine it seems to me that the offending line is:
```
LDSHARED = /blues/gpfs/home/software/spack/lib/spack/env/intel/icc
```
which in my case with this same branch reads:
```
LDSHARED = /home/culpo/github/spack/lib/spack/env/intel/icc -shared -Wl,-soname,libz.so.1,--version-script,zlib.map
```
I was also able to reproduce your error manually doing:
```
$ spack cd zlib%intel
$ spack env zlib%intel bash
$ LDSHARED=~/github/spack/lib/spack/env/intel/icc ./configure
```
Any chance that somehow you have `LDSHARED` or similar variables set in your environment? I believe the cause of this error could be an environment variable interfering with the `configure` tests of `zlib`

**Comment by adamjstewart** [comment]
Unfortunately not:

```bash
[09:41:54] ajstewart@blogin4:/soft/spack (fixes/intel_openmpi)
$ echo $LDSHARED

[09:44:24] ajstewart@blogin4:/soft/spack (fixes/intel_openmpi)
$ spack install zlib %intel
==> Installing zlib
==> Trying to fetch from http://zlib.net/zlib-1.2.8.tar.gz
######################################################################## 100.0%
==> Staging archive: /blues/gpfs/home/software/spack/var/spack/stage/zlib-1.2.8-e7ta6ybpm5e2thhmyem7vfbrokzhki3s/zlib-1.2.8.tar.gz
==> Created stage in /blues/gpfs/home/software/spack/var/spack/stage/zlib-1.2.8-e7ta6ybpm5e2thhmyem7vfbrokzhki3s.
==> No patches needed for zlib.
==> Building zlib.
==> Successfully installed zlib.
  Fetch: 0.61s.  Build: 11.55s.  Total: 12.15s.
[+] /blues/gpfs/home/software/spack/opt/spack/linux-x86_64/intel-16.0.1/zlib-1.2.8-e7ta6ybpm5e2thhmyem7vfbrokzhki3s
[09:44:58] ajstewart@blogin4:/soft/spack (fixes/intel_openmpi)
$ ls /blues/gpfs/home/software/spack/opt/spack/linux-x86_64/intel-16.0.1/zlib-1.2.8-e7ta6ybpm5e2thhmyem7vfbrokzhki3s/lib/
libz.a  pkgconfig
[09:45:42] ajstewart@blogin4:/soft/spack (fixes/intel_openmpi)
$ spack uninstall zlib %intel
==> Successfully uninstalled zlib@1.2.8%intel@16.0.1=linux-x86_64-e7ta6yb.
[09:46:04] ajstewart@blogin4:/soft/spack (fixes/intel_openmpi)
$ git checkout develop
Switched to branch 'develop'
Your branch is up-to-date with 'upstream/develop'.
[09:46:09] ajstewart@blogin4:/soft/spack (develop)
$ echo $LDSHARED

[09:46:12] ajstewart@blogin4:/soft/spack (develop)
$ spack install zlib %intel
==> Installing zlib
==> Trying to fetch from http://zlib.net/zlib-1.2.8.tar.gz
######################################################################## 100.0%
==> Staging archive: /blues/gpfs/home/software/spack/var/spack/stage/zlib-1.2.8-e7ta6ybpm5e2thhmyem7vfbrokzhki3s/zlib-1.2.8.tar.gz
==> Created stage in /blues/gpfs/home/software/spack/var/spack/stage/zlib-1.2.8-e7ta6ybpm5e2thhmyem7vfbrokzhki3s.
==> No patches needed for zlib.
==> Building zlib.
==> Successfully installed zlib.
  Fetch: 0.35s.  Build: 15.48s.  Total: 15.83s.
[+] /blues/gpfs/home/software/spack/opt/spack/linux-x86_64/intel-16.0.1/zlib-1.2.8-e7ta6ybpm5e2thhmyem7vfbrokzhki3s
[09:46:41] ajstewart@blogin4:/soft/spack (develop)
$ ls /blues/gpfs/home/software/spack/opt/spack/linux-x86_64/intel-16.0.1/zlib-1.2.8-e7ta6ybpm5e2thhmyem7vfbrokzhki3s/lib/
libz.a  libz.so  libz.so.1  libz.so.1.2.8  pkgconfig
```

I also tried removing the `-L` flags in my `icc.cfg`, `icpc.cfg`, and `ifort.cfg` files to see if that was the issue, but it didn't help.

**Comment by alalazo** [comment]
@adamjstewart @tgamblin I think the problem here is Intel 16 related. I don't have access to Intel 16 right now, but asking a friend in another site to try with Intel 15 and 16 said that 15 went smooth, while 16 failed badly (in his case at configure time). He didn't try though with the current develop branch and 16.

I'll try to ask our sysadmins if it will be possible to install the latest version of the Intel compiler on site to experiment more. In the meanwhile : @adamjstewart do you have Intel 15 still available on the same machine? If yes could you give it a try?

**Comment by adamjstewart** [comment]
```bash
[10:20:19] ajstewart@blogin4:/soft/spack (fixes/intel_openmpi)
$ spack install zlib %intel@15
==> Installing zlib
==> Trying to fetch from http://zlib.net/zlib-1.2.8.tar.gz
######################################################################## 100.0%
==> Staging archive: /blues/gpfs/home/software/spack/var/spack/stage/zlib-1.2.8-2e5cnaaq4ikyynuuwerltlnsunmoynbn/zlib-1.2.8.tar.gz
==> Created stage in /blues/gpfs/home/software/spack/var/spack/stage/zlib-1.2.8-2e5cnaaq4ikyynuuwerltlnsunmoynbn.
==> No patches needed for zlib.
==> Building zlib.
==> Successfully installed zlib.
  Fetch: 0.74s.  Build: 11.55s.  Total: 12.29s.
[+] /blues/gpfs/home/software/spack/opt/spack/linux-x86_64/intel-15.0.0/zlib-1.2.8-2e5cnaaq4ikyynuuwerltlnsunmoynbn
[10:23:45] /soft/spack (fixes/intel_openmpi)
$ l /blues/gpfs/home/software/spack/opt/spack/linux-x86_64/intel-15.0.0/zlib-1.2.8-2e5cnaaq4ikyynuuwerltlnsunmoynbn/lib/
libz.a  pkgconfig
```

:confused: Am I doing something wrong here?

**Comment by alalazo** [comment]
@adamjstewart I don't think so. I suspect the Intel compiler behavior could be very site dependent. I'll try to read the configure script in zlib and see if the different behavior with respect to the `-v` flag may affect in any way how the linker is identified.

**Comment by luigi-calori** [comment]
already reported in
https://groups.google.com/forum/#!topic/spack/Jj9-iWiMIAU
I can confirm that with the latest develop plus this PR the build:

spack -d --verbose install --jobs 20 --keep-stage openmpi@1.8.8%intel@15.0.2

exit with error

make[2]: Entering directory `/tmp/lcalori0/spack-stage/spack-stage-Gn5DWD/openmpi-1.8.8/ompi/mpi/cxx'
  CXX      mpicxx.lo
  CXX      comm.lo
  CXX      intercepts.lo
  CXX      file.lo
  CXX      datatype.lo
  CXX      win.lo
  CXXLD    libmpi_cxx.la
../../../libtool: eval: line 7623: unexpected EOF while looking for matching `"'
../../../libtool: eval: line 7624: syntax error: unexpected end of file
make[2]: *** [libmpi_cxx.la] Error 2
make[2]: Leaving directory `/tmp/lcalori0/spack-stage/spack-stage-Gn5DWD/openmpi-1.8.8/ompi/mpi/cxx'


this somehow traces back to a spuious -->"<-- being appended by libtool  somehow on his -L dependency.
T be noted that a new intel compiled libtool package has been built due to a dependency in libpciaccess 

Is the dependency on libtool needed as the intel toolchain need an intel-compiled libtool insteas of what provided by the system?

If it is not the case, I would suggest to mate that requirement optional or better site_configurable somehow.

I' ve tested  a workaround,  that patch ltmain.sh files in openmpi source to strip off trailing -->"<--

As I' m in the phase of evaluating spack as a possible replacement of our in-house scripts for installing module base packages I would like to build some packages with the  intel toolchain that is quite heavvily used on our site

**Comment by alalazo** [comment]
@luigi-calori  I'll get back to this asap. I tracked what was the origin of the spurious dependency, i.e. the : 
```
$CC -shared $CFLAGS -v conftest.$objext 2>&1 | $GREP -v "^Configured with:" | $GREP "\-L"
```
test in configure and this PR was actually solving the problem (you may see [here](https://groups.google.com/forum/#!topic/spack/hvo1h9yyhrw) the entire discussion). I still have to investigate why this regression came up.

**Comment by luigi-calori** [comment]
Ok, so I'll wait.
If you can find a solution that is general and not package-specific is surely much better.

Nevertheless, what puzzles me is that other packages do not exibit the problem

I agree that patching the not much robust
openmpi  ltmain.sh code
( it is probably generated from some template as it repeats in different folders)
is not clean even if it probably does not cause much harm

Best

----- Original Message -----

**Comment by alalazo** [comment]
@luigi-calori At the time I submitted this I saw the problem with every package that was using `libtool`.

**Comment by alalazo** [comment]
@adamjstewart @luigi-calori can you check it now? It seems there were a couple of bugs in there...

**Comment by luigi-calori** [comment]
On 10/03/2016 15:14, Massimiliano Culpo wrote:
>
> @adamjstewart <https://github.com/adamjstewart> @luigi-calori 
> <https://github.com/luigi-calori> can you check it now? It seems there 
> were a couple of bugs in there...
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/LLNL/spack/pull/451#issuecomment-194864311>.
>
Wow!! it seem working now
Thanks a lot..

I' ve compiled also hdf5 with

spack install hdf5@1.8.15+mpi%intel@15.0.2^openmpi@1.8.8


-- 
Luigi Calori
SuperComputing Applications and Innovation Department
CINECA - via Magnanelli, 6/3, 40033 Casalecchio di Reno (Bologna) - ITALY
Tel: +39 051 6171509  Fax: +39 051 6132198
hpc.cineca.it

**Comment by tgamblin** [comment]
@luigi-calori, @alalazo: awesome!  Do you think this is ready to merge now, or is there more that needs to be added/more testing needed?

**Comment by adamjstewart** [comment]
Let me try building OpenMPI and zlib. If they work now for me I'll sign off.

**Comment by tgamblin** [comment]
Thanks for all the work on this guys -- sorry I have not been able to help more on our end.

**Comment by adamjstewart** [comment]
@alalazo: your changes also fixed my problems with the zlib installation!
@tgamblin: I'm still waiting on OpenMPI to install (it takes forever with Intel) but it is probably safe to merge this PR now.

**Comment by tgamblin** [comment]
@adamjstewart I can wait :smile:

**Comment by adamjstewart** [comment]
OpenMPI finished installing! Go ahead and merge.

**Comment by tgamblin** [comment]
Thanks everyone!

**Pull request merged**